protocols/v2/binary-sv2/serde-sv2/src/primitives/sequences/option.rs

```diff
@@ -65,11 +68,20 @@ impl<'s, T: Clone + Serialize + TryFromBSlice<'s>> Serialize for Sv2Option<'s, T
                 seq.end()
             }
             (None, Some(data)) => {
-                let tuple = (data.len() as u8, &data[..]);
-                let mut seq = serializer.serialize_tuple(2)?;
-                seq.serialize_element(&tuple.0)?;
-                seq.serialize_element(tuple.1)?;
-                seq.end()
+                if serializer.is_human_readable() {
+                    let data_ = self.data.clone().unwrap();
```

Comment:
Because I had to do this modification several times and I did past and copy, but you are right I should use the `data` in the match branch!
Thank you!